### PR TITLE
feat: recursively include propagatedBuildinputs paths

### PR DESCRIPTION
**Pull request by ysndr**
Read `nix-support/propagated-build-inputs` and add paths to environment.
This will also recursively do the same with already postponed paths.

closes flox/product#565

### CODE REVIEW

**Comment by ysndr** [comment]
Based on the test environment PR #487, hence a draft until that is accepted.
Please **do** review already

File: env-builder/src/buildenv.cc [hunk 1]
```diff
@@ -271,6 +285,7 @@ buildEnvironment( const Path & out, Packages && pkgs )
       postponed.swap( pkgDirs );
       for ( const auto & pkgDir : pkgDirs )
         {
+          printf( "postponed: %s\n", pkgDir.c_str() );
```
Comment (garbas) [inline env-builder/src/buildenv.cc:288]: Is this something you forgot to remove?

**Review by garbas — APPROVED** [review]

File: env-builder/src/buildenv.cc (hunk 1 above)
Comment (ysndr) [reply env-builder/src/buildenv.cc:288]: Yeah, good ol' printf debugging ;)

Comment (ysndr) [reply env-builder/src/buildenv.cc:288]:
```suggestion
```

**Pull request merged**